File: src/main/java/me/lordsaad/modeoff/common/command/CommandTpPlot.java
package me.lordsaad.modeoff.common.command;

import me.lordsaad.modeoff.api.PlotManager;
import net.minecraft.command.CommandBase;
import net.minecraft.command.CommandException;
import net.minecraft.command.ICommandSender;
import net.minecraft.command.PlayerNotFoundException;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.server.MinecraftServer;
import net.minecraft.util.math.BlockPos;
import org.jetbrains.annotations.NotNull;

import javax.annotation.Nullable;
import java.util.Collections;
import java.util.List;

/**
 * Created by LordSaad.
 */
public class CommandTpPlot extends CommandBase {

	@NotNull
	@Override
	public String getName() {
		return "plot_tp";
	}

	@NotNull
	@Override
	public String getUsage(@NotNull ICommandSender sender) {
		return "/plot_tp [username/plotID] [username/plotID]";
	}

	@Override
	public void execute(@NotNull MinecraftServer server, @NotNull ICommandSender sender, @NotNull String[] args) throws CommandException {
		//if (!(sender instanceof EntityPlayer)) {
		//	sender.sendMessage(new TextComponentString(TextFormatting.RED + "You do not have permission to use this command."));
		//	return;
		//}
//
		//if (!CommonProxy.teamMembers.contains(((EntityPlayer) sender).getUniqueID()) || !CommonProxy.contestants.contains(((EntityPlayer) sender).getUniqueID())) {
		//	sender.sendMessage(new TextComponentString(TextFormatting.RED + "You do not have permission to use this command."));
		//	return;
		//}

		if (args.length == 0) {
			new PlotManager(getCommandSenderAsPlayer(sender)).teleportToCenter();

		} else if (args.length == 1) {
			try {
				int plotID = Integer.parseInt(args[0]);
				PlotManager.teleportToPlot(getCommandSenderAsPlayer(sender), new PlotManager(sender.getEntityWorld(), plotID).plotID);
			} catch (NumberFormatException e) {
				PlotManager.teleportToPlot(getCommandSenderAsPlayer(sender), new PlotManager(getPlayer(server, sender, args[0])).plotID);
			}
		} else {
			EntityPlayer player1;
			int id;
			try {
				int plotID = Integer.parseInt(args[0]);
				PlotManager manager = new PlotManager(sender.getEntityWorld(), plotID);
				if (manager.player == null)
					throw new PlayerNotFoundException("Could not find player of plot id " + plotID, sender);
				player1 = manager.player;
			} catch (NumberFormatException e) {
				player1 = getPlayer(server, sender, args[0]);
			}

			PlotManager manager = new PlotManager(getPlayer(server, sender, args[1]));
			if (manager.plotID < 0)
				throw new PlayerNotFoundException("Could not find plot id for player" + args[1], sender);
			id = manager.plotID;

			PlotManager.teleportToPlot(player1, id);
		}
	}

	@NotNull
	@Override
	public List<String> getTabCompletions(MinecraftServer server, ICommandSender sender, String[] args, @Nullable BlockPos targetPos) {
		return args.length >= 1 ? getListOfStringsMatchingLastWord(args, server.getOnlinePlayerNames()) : Collections.emptyList();
	}
}

File: src/main/kotlin/me/lordsaad/modeoff/common/command/CommandTpPlot.kt
package me.lordsaad.modeoff.common.command

import me.lordsaad.modeoff.api.PlotAssigningManager
import me.lordsaad.modeoff.api.PlotManager
import net.minecraft.command.*
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.server.MinecraftServer
import net.minecraft.util.math.BlockPos

/**
 * Created by LordSaad.
 */
class CommandTpPlot : CommandBase() {

    override fun getName() = "plot_tp"

    override fun getUsage(sender: ICommandSender) = "/plot_tp [username/plotID] [username/plotID]"

    @Throws(CommandException::class)
    override fun execute(server: MinecraftServer, sender: ICommandSender, args: Array<String>) {
        //if (!(sender instanceof EntityPlayer)) {
        //	sender.sendMessage(new TextComponentString(TextFormatting.RED + "You do not have permission to use this command."));
        //	return;
        //}
        //
        //if (!CommonProxy.teamMembers.contains(((EntityPlayer) sender).getUniqueID()) || !CommonProxy.contestants.contains(((EntityPlayer) sender).getUniqueID())) {
        //	sender.sendMessage(new TextComponentString(TextFormatting.RED + "You do not have permission to use this command."));
        //	return;
        //}

        when (args.size) {
            0 -> {
                val player = CommandBase.getCommandSenderAsPlayer(sender)
                PlotManager(player).teleportPlayerToCenter(player)
            }
            1 -> {
                val (plot) = args
                val plotManager = getPlotManagerByPlotIdOrPlayer(server, sender, plot)
                PlotManager.teleportToPlot(CommandBase.getCommandSenderAsPlayer(sender), plotManager.plotID)
            }
            2 -> {
                val (player, plot) = args
                val targetPlayer = CommandBase.getPlayer(server, sender, player)
                val plotManager = getPlotManagerByPlotIdOrPlayer(server, sender, plot)

                PlotManager.teleportToPlot(targetPlayer, plotManager.plotID)
            }
            else -> throw WrongUsageException(getUsage(sender))
        }
    }

    private fun getPlotManagerByPlotIdOrPlayer(server: MinecraftServer, sender: ICommandSender, plotIdOrPlayer: String): PlotManager {
        return try {
            val plotID = Integer.parseInt(plotIdOrPlayer)
            PlotManager(PlotAssigningManager.getUUIDForPlot(plotID)!!, plotID)
        } catch (e: NumberFormatException) {
            PlotManager(CommandBase.getPlayer(server, sender, plotIdOrPlayer))
        }
    }

    override fun getTabCompletions(server: MinecraftServer, sender: ICommandSender, args: Array<String>, targetPos: BlockPos?): List<String> {
        return if (args.isNotEmpty()) CommandBase.getListOfStringsMatchingLastWord(args, *server.onlinePlayerNames) else emptyList<String>()
    }
}
